### PR TITLE
Drop support for EOL Python 2

### PR DESCRIPTION
**Pull request by a-detiste**
python3-future is broken under Python 3.12 and being removed from Debian

python3-unittest2 is another ancient compatibility layer also slated for removal

greetings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,6 @@ $ pip install mpegdash
 ## Test
 
 ```bash
-$ python -m unittest discover
```
Comment (rlaphoenix) [inline README.md:22]: I think remove the other line, since `python3` is also somewhat an ancient concept nowadays.

File: mpegdash/utils.py [hunk 2]
```diff
@@ -19,7 +18,7 @@ def parse_child_nodes(xmlnode, tag_name, node_type):
 
     nodes = []
     for elem in elements:
-        if node_type in (unicode, str):
```
Comment (rlaphoenix) [inline mpegdash/utils.py:22]: `if isinstance(node_type, str):` is best

**Comment by a-detiste** [comment]
Thank you for the review

**Comment by a-detiste** [comment]
Ping

**Comment by sangwonl** [comment]
OK, I agree to drop support for python2. Thanks.

(Those who want to use python-mpegdash in python2 environment will still be able to use 0.3.1 without any major issue.)

**Comment by sangwonl** [comment]
need to drop 2.7 in github action workflow too (`.github/workflows/build-status.yml`)

**Review by sangwonl — APPROVED** [review]

**Pull request merged**